app/models/job_completion_summary.rb

```diff
@@ -63,10 +33,21 @@ def definition_name
     end
   end
 
+  def job_completions
+    JobCompletion.where(source_id: source_id, process_type: process_type, job_type: job_type)
+  end
+
+  def last_completed_at
+    job_completions.order(updated_at: :desc).first&.updated_at
+  end
+
+  def increment_completion_count
+    update(completion_count: completion_count + 1)
```

Comment:
Do we need this with a has_many belongs_to relationship? A simple count on the relationship would do the job without putting pressure on MySQL